backend/backend/core/tools.py

```diff
@@ -32,7 +33,13 @@ def __init__(self):
 
     async def ainvoke_tool_call_artifact(self, query):
         results = await self.ainvoke(
```

Comment:
The 'ainvoke' method is called with a dictionary that has a 'type' key set to 'tool_call'. This could be a potential security issue if the tool being called has access to sensitive information.